### PR TITLE
Added recipe for cmd2

### PR DESCRIPTION
**Pull request by pmlandwehr**
[`cmd2`](https://pypi.python.org/pypi/cmd2) expands the functions of the standard library's `cmd module.  Various openstack python-based CLI tools use `cmd2` as a component; this is a jumping-off point for adding new openstack modules to conda-forge.

Pinging @catherinedevlin in case she would like to be added as a maintainer to the `cmd2` builder on conda-forge, a library of community-build conda packages. The maintenance burden here is light: all testing and releases are built and deployed automatically with CI. Changes to how the package is being built can be controlled by changing the recipe which will be located at https://github.com/conda-forge/cmd2-feedstock shortly after this PR is finished and merged. If you aren't interested in helping maintain the build, that's entirely fine too.

### CODE REVIEW

**Comment by pmlandwehr** [comment]
A note - I'm currently skipping the python 3.5 build because `import cmd2` was crashing out with this error:
`AttributeError: module 'subprocess' has no attribute 'mswindows'`
This has been reported [here](https://bitbucket.org/catherinedevlin/cmd2/issues/18/python-35-renames-subprocessmswindows), but the supplied patch file doesn't seem to work on OS X.

File: recipes/cmd2/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  {{ hash_type }}: {{ hash_val }}
+  # patches:
+  #   - python35-compat.patch  # [py35]
```
Comment (ocefpaf) [inline recipes/cmd2/meta.yaml]: @pmlandwehr are you working on a patch or is this good to go?

**Comment by pmlandwehr** [comment]
I'm going to double-check on the patch, then call it good-to-go. This is a component of other opencpu stuff, so I'd like to get it fixed.

**Comment by jakirkham** [comment]
Always possible to try patching in the feedstock.

File: recipes/cmd2/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+build:
+  skip: True  # [py35]
```
Comment (ocefpaf) [inline recipes/cmd2/meta.yaml]: This is fine. Just open an issue in the feedstock to remind you to remove the `Skip` once you patch or update.

**Pull request merged**